### ISSUE TITLE
A11y:  Calcite block header uses h3 for heading content

### ISSUE THREAD

**Issue by kellyhutchins** [Esri/calcite-components#1531]
### Summary
If an app uses the calcite block the heading section renders as h3 and there doesn't seem to be a documented way to overwrite that behavior. If the app doesn't have an h2 then this will be flagged as an accessibility issue. There doesn't seem to be a slot for heading here: https://github.com/Esri/calcite-components/tree/master/src/components/calcite-block
### Actual Behavior
The app should be able to overwrite the h tag associated with the header section to fit the apps structure. 
### Reproduction Steps

1. This app has an h1 but no h2 so when you test the accessibility using a tool like axe the h3 headings in the calcite-block are flagged as issues. 
2. https://catsqa.mapsdevext.arcgis.com/apps/instant/minimalist/index.html?appid=23d56c24f3ba4577a9f5abe452acaebf

**Comment by macandcheese** [comment]
Keep open

**Comment by jcfranco** [comment]
@kellyhutchins Could you help verify this one? 🙇🏼‍♂️

**Comment by kellyhutchins** [comment]
@jcfranco yes I'll test this today

**Comment by kellyhutchins** [comment]
Looks good!

**Comment by jcfranco** [comment]
Awesome. Thanks for verifying, @kellyhutchins!
